Software/smart-charger/src/pages/Register/Register.jsx

```diff
@@ -0,0 +1,212 @@
+import {
+  ButtonWrapper,
+  RegisterError,
+  RegisterForm,
+  RegisterFormError,
+  RegisterFormField,
+  RegisterFormLabel,
+  RegisterFormLink,
+  RegisterFormText,
+  RegisterWrapper,
+} from "./RegisterStyles";
+import TextField from "../../components/TextField/TextField";
+import Button from "../../components/Button/Button";
+import { useNavigate } from "react-router";
+import {
+  loginUser as login,
+  registerUser as register,
+} from "../../utils/api/users";
+import { AuthContext } from "../../context/AuthContext";
+import { decodeToken } from "react-jwt";
+
+export const Register = () => {
+  const navigate = useNavigate();
+  const [error, setError] = useState("");
+  const { setIsLoggedIn, setRole } = useContext(AuthContext);
+
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+
+  const [firstNameError, setFirstNameError] = useState("");
+  const [lastNameError, setLastNameError] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+
+  const registerUser = async () => {
+    if (
+      validateFirstName() &&
+      validateLastName() &&
+      validateEmail() &&
+      validatePassword() &&
+      validateConfirmPassword()
+    ) {
+      let res = await register({
+        firstName: firstName,
+        lastName: lastName,
+        email: email,
+        password: password,
+      });
+      if (!res.success) {
+        setError(res.message);
+      } else {
+        await loginUser({
+          email: email,
+          password: password,
+        });
+      }
+    }
+  };
+
+  const loginUser = async (user) => {
+    const res = await login(user);
+    if (res.success) {
+      localStorage.setItem("jwt", res.token);
+      const jwtData = decodeToken(res.token);
+      setRole(jwtData.roleId);
+      setIsLoggedIn(true);
+    } else {
+      setError(res.message);
+    }
+  };
+
+  const validateFirstName = () => {
+    if (firstName.length === 0) {
+      setFirstNameError("First name can't be empty!");
+      return false;
+    } else {
+      setFirstNameError("");
+      return true;
+    }
+  };
+
+  const validateLastName = () => {
+    if (lastName.length === 0) {
+      setLastNameError("Last name can't be empty!");
+      return false;
+    } else {
+      setLastNameError("");
+      return true;
+    }
+  };
+
+  const validateEmail = () => {
+    // const emailRegex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/;
+    const emailRegex = /^[^@\s]+@[^@\s]+\.[^@\s]+$/i;
+    if (!emailRegex.test(email)) {
+      setEmailError("Email is invalid!");
+      return false;
+    } else {
+      setEmailError("");
+      return true;
+    }
+  };
+
+  const validatePassword = () => {
```

Comment:
Like i mentioned for "validateFirstName" you could implement check for whitespaces in "validateEmail" and "validatePassword" soo that we can avoid having users with whitespaces as names.

Reply:
I added the trim function here as well.